### PR TITLE
1.30.0 release blogpost

### PR DESCRIPTION
**Pull request by AkiraBrand**
First draft of 1.30.0 release blogpost

### CODE REVIEW

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+{
+  "claims": {
+     "roles": [ "awesome" ],
+     "id": 42
+   }
+}
+```
+
+Using this ability, a new user on your app can start interacting with the app as a unique user, before needing to create an account. Then, once they do create an account, their unique Id is translated onto their profile.
```
Comment (jobannon) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:40]: With this feature, an anonymous app visitor can interact with your platform and subsequently create an account and have all those interactions recorded (if desired).  

A common use case might be a visitor and a shopping cart.  You would not want to lose precious user browsing data while the user was evaluating your site.  Once they commit to a purchase and register, likely you will want to preserve portions of their initial visit to generate future revenue.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     "id": 42
+   }
+}
+```
+
+Using this ability, a new user on your app can start interacting with the app as a unique user, before needing to create an account. Then, once they do create an account, their unique Id is translated onto their profile. 
+
+## Threat Detection Feature
+
+Flagging and responding to suspicious behaviour is a huge advantage in any cybersecurity product, and we wanted to bring our customers as many advantages as we could in this arena. So, we created a large feature that has best-practice funtionality built in to deal with any sort of bizarre behaviours around user creation, user updates, and IP location and fingerprinting.
```
Comment (jobannon) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:44]: Preference: 
Flagging and responding to suspicious behavior is a huge advantage in any cybersecurity product, so naturally, we incorporated this functionality into the ever-evolving FusionAuth platform.

Comment (jobannon) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:44]: So, we created a large feature that has best-practice functionality built in to deal with any sort of bizarre behaviors around user creation, user updates, and IP location and fingerprinting.

This sentence seems vague.
_maybe_?
In this release of FusionAuth, we use IP location data, user fingerprinting, and other indicators around user events to signal suspicious behavior.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+ - Send two-factor
+ 
+Capcha - Uses additonal (re)captcha to guard against brute force attacks. 
+- Adds a (re)captcha for login pages which use an email/password combination
+- Adds support for Google ReCaptcha v2, Google ReCaptcha v3, HCaptcha and Hcaptcha Enterprise
+
+IP Location Services
+- Includes the IP address of the geographic location where a Forgot/Reset Password email was created, so the recipient can identify if the location is genuinely their request or not. 
+- Flags suspicious IP adddresses on login and sends an email to notify the user of a new login with the IP's approximate location
+- Calculates 'impossible travel' to see if a user could genuinely login in different locations around the globe in a reasonable time frame. (This does not provide support if your user is a quantum being...)
```
Comment (jobannon) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:63]: user could genuinely login -> user could realistically login

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+- With unexpected IP addresses, sends a user an email to notify them of a new login with an approximate location of the IP address. 
+
+To learn more, you can take a look at our [release notes](/docs/v1/tech/release-notes/#version-1-30-0). 
+
+## Webhook Events
+
+We've added 38 webhook events to help you moniter what is going on with your app.  These webhooks cover an entire breadth of createing, updated, and deleting a user, as well as the user registration and password reset lifecycle. Lastly, we've added some webhook events to moniter any kickstart files you may create, as well as moniter JWT behavior with refresh tokens and public/private keypairs are updated.
```
Comment (jobannon) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:70]: moniter spelling

Comment (jobannon) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:70]: createing

Comment (jobannon) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:70]: updated

Comment (jobannon) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:70]: If you read this aloud, this sentence does not flow
Lastly, we've added some webhook events to moniter any kickstart files you may create, as well as moniter JWT behavior with refresh tokens and public/private keypairs are updated.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+## Webhook Events
+
+We've added 38 webhook events to help you moniter what is going on with your app.  These webhooks cover an entire breadth of createing, updated, and deleting a user, as well as the user registration and password reset lifecycle. Lastly, we've added some webhook events to moniter any kickstart files you may create, as well as moniter JWT behavior with refresh tokens and public/private keypairs are updated. 
+
+## The rest of it
+
+Some of the other enhancements and fixes included in this release:
+
+* search on `oldValue`, `newValue`, and `reason` in the Audit Log
+* More themed pages have access to the currently logged in user using the `currentUser` varible
```
Comment (jobannon) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:77]: varible

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+## Upgrade at will
+
+The [release notes](/docs/v1/tech/release-notes/#version-1-30-0) are a guide of the changes, fixes, and new features. Please read them carefully to see if any features you use have been modified.
+
+If you'd like to upgrade your self-hosted FusionAuth instance, see our [upgrade guide](/docs/v1/tech/installation-guide/upgrade/). 
+
+If you have a FusionAuth Cloud deployment, proceed to the "Deployments" tab on your [account dashboard](https://account.fusionauth.io/account/deployment/){:target="_blank"} and upgrade your servers. If you have any questions about the upgrade, [please open a support ticket](https://account.fusionauth.io/account/support/){:target="_blank"}.
+
+Or, if we've piqued your interest and you'd like to use FusionAuth, [check out your options](/pricing/).
```
Comment (jobannon) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:89]: I am not sure why this link looks strange to me 
[check out your options](/pricing/).

could be fine...

**Review by jobannon — COMMENTED** [review]
A few thoughts.
I would run this through Grammarly to capture some spelling adjustments.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+This release contained a number of features, enhancements, and bug fixes. Please see the [release notes](/docs/v1/tech/release-notes/#version-1-30-0) for a full breakdown of the changes between 1.29 and 1.30.0. 
+
+There are a few items worth calling out.
+
+## JWT Vending Machine
+
+There are scenarios where a user does not yet exist, or is in the process of being created and needs access to services. This may be an anonymous user, or a future user. In these cases we cannot yet authenticate to FusionAuth because the user does not yet exist, or they may not yet have a password.
+
+To solve this, we have created the possiblity of opening an API that allows a JWT to be created with a payload defined by the API called.  This would require an API key, so the caller must be privledged.
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:24]:
```suggestion
To address this, the new release includes an API allowing a JWT to be created without a corresponding user. 
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+There are a few items worth calling out.
+
+## JWT Vending Machine
+
+There are scenarios where a user does not yet exist, or is in the process of being created and needs access to services. This may be an anonymous user, or a future user. In these cases we cannot yet authenticate to FusionAuth because the user does not yet exist, or they may not yet have a password.
+
+To solve this, we have created the possiblity of opening an API that allows a JWT to be created with a payload defined by the API called.  This would require an API key, so the caller must be privledged.  
+
+Here is an example of the endpoint at work:
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:26]: I found this example confusing and not really clear. Maybe offer a sample curl script with the input (which you provide) and the output. Or you could omit the curl script and just say something like "when you call the endpoint with the below input: <show input>, you'll get back a signed JWT, which, when decoded and validated, will look like the below output: <show output>"

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 1 above)
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:40]:
```suggestion
Using this ability, a new user on your app can start interacting with the app as a unique user, before needing to create an account. Then, once they do create an account, the unique Id can be used to ensure any saved profile data is retained. 
```

This isn't something automatic, right? They need to do this themselves by creating a user with the given subject Id?

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+     "roles": [ "awesome" ],
+     "id": 42
+   }
+}
+```
+
+Using this ability, a new user on your app can start interacting with the app as a unique user, before needing to create an account. Then, once they do create an account, their unique Id is translated onto their profile. 
+
+## Threat Detection Feature
+
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:43]: Please mention that this is a paid feature, available to enterprise clients only. You could also mention that turning this on requires opening a support ticket, as we're rolling this out over time.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 2 above)
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:44]:
```suggestion
Flagging and responding to suspicious behavior is a part of any cybersecurity product, and we wanted to bring our customers as many advantages as we could in this arena. So, we created advanced threat detection, a feature that brings best-practice functionality to help you deal with bizarre, possibly malicious behavior around logins, registrations, user creation, and user updates.
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+Rate limiting per user for specific requests such as:
+ - Failed login
+ - Forgot password
+ - Send email verification
+ - Send passwordless
+ - Send registration verification
+ - Send two-factor
+ 
+Capcha - Uses additonal (re)captcha to guard against brute force attacks.
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:56]: Capcha -> Captcha

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+}
+```
+
+Using this ability, a new user on your app can start interacting with the app as a unique user, before needing to create an account. Then, once they do create an account, their unique Id is translated onto their profile. 
+
+## Threat Detection Feature
+
+Flagging and responding to suspicious behaviour is a huge advantage in any cybersecurity product, and we wanted to bring our customers as many advantages as we could in this arena. So, we created a large feature that has best-practice funtionality built in to deal with any sort of bizarre behaviours around user creation, user updates, and IP location and fingerprinting. 
+
+While the entirety of this feature is too large to cover here, we want to call out a few notable aspects:
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:46]:
```suggestion
While the entirety of this feature is too large to cover here, we want to call out a few notable aspects.
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+
+Rate limiting per user for specific requests such as:
+ - Failed login
+ - Forgot password
+ - Send email verification
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:51]: Would probably pick 3 and say "and more" for the 4th bullet point.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 10 above)
Comment (mooreds) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:56]: I'm not clear on what the difference between captcha and (re)captcha is? Maybe only use the former?

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+ - Send passwordless
+ - Send registration verification
+ - Send two-factor
+ 
+Capcha - Uses additonal (re)captcha to guard against brute force attacks. 
+- Adds a (re)captcha for login pages which use an email/password combination
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:57]:
```suggestion
- Adds a (re)captcha for login pages
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+- Adds support for Google ReCaptcha v2, Google ReCaptcha v3, HCaptcha and Hcaptcha Enterprise
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:58]:
```suggestion
- Includes support for Google ReCaptcha v2, Google ReCaptcha v3, HCaptcha and Hcaptcha Enterprise
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+
+IP Location Services
+- Includes the IP address of the geographic location where a Forgot/Reset Password email was created, so the recipient can identify if the location is genuinely their request or not.
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:61]:
```suggestion
- Includes the geographic location where a Forgot/Reset Password email was requested, so the recipient can identify if the location is genuinely their request or not. 
```

I haven't looked at this closely, but isn't it the geographic location, not the IP address, that is included?

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 3 above)
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:63]:
```suggestion
- Calculates 'impossible travel' to see if a user could genuinely login at different locations around the globe in a reasonable time frame. (This does not provide support if your user is a quantum being...)
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+ 
+Capcha - Uses additonal (re)captcha to guard against brute force attacks. 
+- Adds a (re)captcha for login pages which use an email/password combination
+- Adds support for Google ReCaptcha v2, Google ReCaptcha v3, HCaptcha and Hcaptcha Enterprise
+
+IP Location Services
+- Includes the IP address of the geographic location where a Forgot/Reset Password email was created, so the recipient can identify if the location is genuinely their request or not. 
+- Flags suspicious IP adddresses on login and sends an email to notify the user of a new login with the IP's approximate location
+- Calculates 'impossible travel' to see if a user could genuinely login in different locations around the globe in a reasonable time frame. (This does not provide support if your user is a quantum being...)
+- With unexpected IP addresses, sends a user an email to notify them of a new login with an approximate location of the IP address.
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:64]:
```suggestion
- When a login request occurs from an unexpected IP addresses, sends a user an email to notify them of a new login with an approximate location of the IP address. 
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 17]
```diff
@@ -0,0 +1,89 @@
...
+ - Send email verification
+ - Send passwordless
+ - Send registration verification
+ - Send two-factor
+ 
+Capcha - Uses additonal (re)captcha to guard against brute force attacks. 
+- Adds a (re)captcha for login pages which use an email/password combination
+- Adds support for Google ReCaptcha v2, Google ReCaptcha v3, HCaptcha and Hcaptcha Enterprise
+
+IP Location Services
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:60]:
```suggestion
Location aware security
```

maybe? Not sure, but for many folks, the location info will be more important than how we get it.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 9 above)
Comment (mooreds) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:43]: Or you could mention this at the end of this section too. See https://fusionauth.io/blog/2021/05/21/announcing-fusionauth-1-27/ for an example. ("This is a reactor feature, requiring a paid license")

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 4 above)
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:70]:
```suggestion
We've added 38 webhook events to help you monitor what is going on with your app and integrate with other systems.  These webhooks cover the entire breadth of creating, updated, and deleting a user, as well as the user registration and password reset lifecycle. 

Lastly, we've added some webhook events to moniter any kickstart files you may create, as well as moniter JWT behavior with refresh tokens and public/private keypairs are updated. 
```

monitor is used a lot, is there another word you can use?

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+- Flags suspicious IP adddresses on login and sends an email to notify the user of a new login with the IP's approximate location
+- Calculates 'impossible travel' to see if a user could genuinely login in different locations around the globe in a reasonable time frame. (This does not provide support if your user is a quantum being...)
+- With unexpected IP addresses, sends a user an email to notify them of a new login with an approximate location of the IP address. 
+
+To learn more, you can take a look at our [release notes](/docs/v1/tech/release-notes/#version-1-30-0). 
+
+## Webhook Events
+
+We've added 38 webhook events to help you moniter what is going on with your app.  These webhooks cover an entire breadth of createing, updated, and deleting a user, as well as the user registration and password reset lifecycle. Lastly, we've added some webhook events to moniter any kickstart files you may create, as well as moniter JWT behavior with refresh tokens and public/private keypairs are updated. 
+
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:71]:
```suggestion
We're working on the documentation for these webhooks, which will be available at <link> shortly.

```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+## The rest of it
+
+Some of the other enhancements and fixes included in this release:
+
+* search on `oldValue`, `newValue`, and `reason` in the Audit Log
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:76]:
```suggestion
* Search on `oldValue`, `newValue`, and `reason` in the Audit Log
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 5 above)
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:77]:
```suggestion
* More themed pages have access to the currently logged in user with the `currentUser` variable.
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+* More themed pages have access to the currently logged in user using the `currentUser` varible
+* Using an IdP with a linking mode other than Create Pending Link, the token is now stored correctly in the Identity Provider Link
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:78]:
```suggestion
* When an IdP has a linking mode other than Create Pending Link, the token is now stored correctly in the Identity Provider Link object.
```

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+
+To learn more, you can take a look at our [release notes](/docs/v1/tech/release-notes/#version-1-30-0). 
+
+## Webhook Events
+
+We've added 38 webhook events to help you moniter what is going on with your app.  These webhooks cover an entire breadth of createing, updated, and deleting a user, as well as the user registration and password reset lifecycle. Lastly, we've added some webhook events to moniter any kickstart files you may create, as well as moniter JWT behavior with refresh tokens and public/private keypairs are updated. 
+
+## The rest of it
+
+Some of the other enhancements and fixes included in this release:
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:74]: If these are sentences (and it seems they are) please capitalize the first letter and add a period at the end.

Comment (mooreds) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:74]: If they aren't, capitalize the first letter and no period.

Comment (mooreds) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:74]: But have it be consistent.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 22]
```diff
@@ -0,0 +1,89 @@
...
+
+* search on `oldValue`, `newValue`, and `reason` in the Audit Log
+* More themed pages have access to the currently logged in user using the `currentUser` varible
+* Using an IdP with a linking mode other than Create Pending Link, the token is now stored correctly in the Identity Provider Link
+* Making an API request to `/api/two-factor/login` with an empty JSON body, an exception now occurs instead of returning a validation error with a `400` status code.
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:79]: I think you want to indicate that this has been fixed :)

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md [hunk 23]
```diff
@@ -0,0 +1,89 @@
...
+ - Forgot password
+ - Send email verification
+ - Send passwordless
+ - Send registration verification
+ - Send two-factor
+ 
+Capcha - Uses additonal (re)captcha to guard against brute force attacks. 
+- Adds a (re)captcha for login pages which use an email/password combination
+- Adds support for Google ReCaptcha v2, Google ReCaptcha v3, HCaptcha and Hcaptcha Enterprise
+
```
Comment (mooreds) [inline site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md:59]: Perfect place for a screenshot of the map. People love pictures.

**Review by mooreds — CHANGES_REQUESTED** [review]
A few changes.

Bigger picture issues:

* please run through google's spellcheck (or another one), there were a few typos

* please read aloud? always a good idea to see if there are incomplete or run-on sentences, or even just unclear concepts.

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 1 above)
Comment (AkiraBrand) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md]: checking now

File: site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md (hunk 15 above)
Comment (AkiraBrand) [reply site/_posts/2021-08-18-annnoucing-fusionauth-1.30.md]: also checking

**Review by mooreds — APPROVED** [review]

**Pull request merged**